### PR TITLE
US126194 Add sample hypermedia data, and extracted out the list item component. 

### PR DESCRIPTION
**Pull request by josephsaffron**
added test data, demo data, and tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,8 +4,10 @@
   "description": "",
   "main": "index.js",
   "dependencies": {
+    "@brightspace-hmc/foundation-components": "git://github.com/BrightspaceHypermediaComponents/foundation-components.git",
     "@brightspace-ui/core": "^1.118.2",
-    "lit-element": "^2.4.0"
+    "lit-element": "^2.4.0",
+    "siren-sdk": "github:BrightspaceHypermediaComponents/siren-sdk#semver:^1"
```
Comment (josephsaffron) [inline package.json:10]: can probably get rid of this

File: src/components/d2l-activity-question-points.js [hunk 2]
```diff
@@ -4,13 +4,17 @@ import '@brightspace-ui/core/components/list/list-item.js';
 import '@brightspace-ui/core/components/list/list-item-content.js';
```
Comment (josephsaffron) [inline src/components/d2l-activity-question-points.js:4]: ignore this file. =/

File: src/components/d2l-activity-question-usage.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		}
+		` ];
+	}
+
+	_onInputChange(e) {
+		const points = e.currentTarget.value;
+		this._setPoints.commit(
+			{
+				points: {
+					observable: observableTypes.property,
```
Comment (josephsaffron) [inline src/components/d2l-activity-question-usage.js:49]: @TomMichielsD2L this is the part we were missing.

File: src/components/d2l-activity-question-usage.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+					value: points
+				}
+			}
+		);
+		this._state.push();
```
Comment (josephsaffron) [inline src/components/d2l-activity-question-usage.js:54]: @TomMichielsD2L and this is the part that does the patch!

File: test/d2l-activity-question-points.test.js [hunk 5]
```diff
@@ -56,24 +56,14 @@ describe('d2l-activity-question-points', () => {
...
-		it('Displays no questions', async() => {
-			setupTestData({
-				questions: []
-			});
-
-			const el = await fixture(defaultFixture);
-			const rows = el.shadowRoot.querySelectorAll('d2l-list-item');
-			expect(rows.length).to.equal(0);
-		});
-
```
Comment (louiscoste) [inline test/d2l-activity-question-points.test.js:68]: Why did you remove this test?

File: test/d2l-activity-question-points.test.js [hunk 6]
```diff
@@ -56,24 +56,14 @@ describe('d2l-activity-question-points', () => {
...
 		it('Displays all questions', async() => {
 			setupTestData({
 				questions: defaultQuestions
 			});
 
 			const el = await fixture(defaultFixture);
 			const rows = el.shadowRoot.querySelectorAll('d2l-list-item');
-			expect(rows.length).to.equal(3);
+			expect(rows.length).to.equal(4);
```
Comment (louiscoste) [inline test/d2l-activity-question-points.test.js:66]: How did this value change if you didn't change the mock data.

File: test/d2l-activity-question-points.test.js [hunk 7]
```diff
@@ -87,14 +77,14 @@ describe('d2l-activity-question-points', () => {
 			fixtureCleanup();
 		});
 
-		it('Button is disabled if invalid field', async() => {
+		it.only('Button is disabled if invalid field', async() => {
```
Comment (louiscoste) [inline test/d2l-activity-question-points.test.js:80]: Is this staying?

File: src/components/d2l-activity-question-usage.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+	render() {
+		return html`
+		${this.points}
```
Comment (louiscoste) [inline src/components/d2l-activity-question-usage.js:75]: Did you mean to keep this?

File: test/d2l-activity-question-points.test.js (hunk 6 above)
Comment (josephsaffron) [reply test/d2l-activity-question-points.test.js:66]: the mock data is now coming from hypermedia entities

File: test/d2l-activity-question-points.test.js (hunk 5 above)
Comment (josephsaffron) [reply test/d2l-activity-question-points.test.js:68]: all these tests need to be rewritten to handle hypermedia

**Review by louiscoste — APPROVED** [review]

File: demo/question-usage.html [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+			<div class="actions">
+				<button>focus</button>
+			</div>
+
+			<d2l-demo-snippet>
+				<template>
+					<d2l-activity-question-usage
+						href="data/d2l-activity-question-usage-chained.json"
+						token="eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6ImIxMTY0ZjA0LWQwYmEtNGVjMS1iYWU3LWM1NWUwZTFiYzJkMCJ9.eyJ0ZW5hbnRpZCI6ImRmZGI3ZTE1LTczNTUtNDU1NC1hOWE1LWRiZjZiOGFlZTZlNyIsInNjb3BlIjoiKjoqOioiLCJqdGkiOiI5OGM3Yzc0Ni05NmUxLTQ2YTItYjlmNi1iZGNiMzQ5YWVkMTEiLCJpc3MiOiJodHRwczovL2FwaS5icmlnaHRzcGFjZS5jb20vYXV0aCIsImF1ZCI6Imh0dHBzOi8vYXBpLmJyaWdodHNwYWNlLmNvbS9hdXRoL3Rva2VuIiwiZXhwIjoxNjE2NjE1MTU2LCJuYmYiOjE2MTY2MTE1NTZ9.OOQUz0UXMFaezbOLuDPHTu2vvvbjsUz4ZHtK_KLNAgMylKP7cWeDDqqmiYAqlmO58NlpaeiEJEnjvOQbh9IeLxRgtlpHB4o4CeKbTF_k2ZFcmaPPDpFqnw61e98sVqxxqvQrNLYiWk2bE3UflmRR1XF5DHWwVSr52X3Wb4UYbWuEh2CvzrMmFidGDFM8ZhOKNHHedrTO8WLgXxwXumr5p-IMUev1LiPTeatKrjB6mPwwG8Hl09tCod8JRocj52KvpbnVbjYQ1yIboO5m8t2fjyuy-TbaoAAhfC-SQJXrf-16e0aWgI5gFbkLPB7S558izadr7vtPYgNnAauoanUSXw">
```
Comment (TomMichielsD2L) [inline demo/question-usage.html:35]: Is the demo actually dependent on the token, can we get rid of it?

File: src/components/d2l-activity-question-usage.js [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+			</d2l-list-item-content>
+			<div class="activity_list__points_input" slot="actions">
+				<label for="points_input_${this.id}" class="points_input__label d2l-label-text">
+					${this.localize('inputLabelPoints')}
+				</label>
+				<d2l-input-number
+					id="points_input_${this.id}"
+					label=${this.localize('inputLabelPoints')}
+					value=${ this.points }
+					@change="${this._onInputChange}"
```
Comment (TomMichielsD2L) [inline src/components/d2l-activity-question-usage.js:96]: Do we still have validation with this change event being tied to a different function?

**Review by TomMichielsD2L — APPROVED** [review]

File: demo/question-usage.html (hunk 9 above)
Comment (josephsaffron) [reply demo/question-usage.html:35]: we can get rid of it.

File: src/components/d2l-activity-question-usage.js (hunk 10 above)
Comment (josephsaffron) [reply src/components/d2l-activity-question-usage.js:96]: we have UI validation, but not the stuff that @louiscoste wrote that checks if all the fields are valid to disable or enable the button

File: demo/question-usage.html (hunk 9 above)
Comment (josephsaffron) [inline demo/question-usage.html:35]:
```suggestion
						token="anytoken">
```

File: src/components/d2l-activity-question-usage.js (hunk 10 above)
Comment (TomMichielsD2L) [reply src/components/d2l-activity-question-usage.js:96]: Should be easy enough to cross that bridge when we get there.
